jax/_src/core.py

```diff
@@ -1729,6 +1732,8 @@ def __eq__(self, other):
     if isinstance(other, DArray) and self._aval == other._aval:
       return self._data == other._data
     return False
+  def __len__(self):
```

Comment:
Do we need this? If not, can we remove it?

Reply:
Actually, we seem to -- the `segment_lengths` can be a `DArray`, and we query its `len`.